### PR TITLE
Rename Cajita -> Cabana::Grid

### LINKED ISSUE #659: Plan to replace Cajita with Cabana::Grid
The current subpackage will be renamed and the namespace aliased with a deprecation tag. Discussion welcome

### PR DESCRIPTION
**Pull request by streeve**
Deprecate Cajita namespace and library. Also deprecate cabanacore->Core for consistency. Closes #659 

To do: 
- [x] Make sure everything has a deprecation tag
- [x] Rename files
- [x] grep for Cajita
- [x] Verify with external builds
- [x] Also update `cabanacore`->`Core` and deprecate
- [x] Pin issue linking here, showing how to update code, and showing how to disable warnings in user builds

Post-merge
- Update wiki

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -239,12 +239,18 @@ endif()
 
 add_subdirectory(core)
 
-option(Cabana_ENABLE_CAJITA "Build Cajita (grid and particle-grid)" ${Cabana_ENABLE_MPI})
-if(Cabana_ENABLE_CAJITA)
+option(Cabana_ENABLE_GRID "Build grid and particle-grid capabilities" ${Cabana_ENABLE_MPI})
+option(Cabana_ENABLE_CAJITA "Build grid and particle-grid capabilities" OFF)
+if (Cabana_ENABLE_CAJITA)
+  message(DEPRECATION "Option Cabana_ENABLE_CAJITA is deprecated, replaced by Cabana_ENABLE_GRID")
+  set(Cabana_ENABLE_GRID ON)
```
Comment (junghans) [inline CMakeLists.txt:246]: The more cmake'ish way would be to use `cmake_dependent_option()`, but as we will ultimately remove this block, I am not it is worth it!

Comment (streeve) [reply CMakeLists.txt:246]: Good to know!

**Comment by streeve** [comment]
Can someone try to build any downstream code with this branch? I'm still having issues with codes linking to `Cabana::Cajita` (even though it's aliased to the new `Cabana::Grid`) @junghans @lebuller @kwitaechong @abisner

**Comment by abisner** [comment]
The `Cajita::isNode`, `Cajita::isCell` type-checker deprecations are missing in the new `Cabana_Grid_Types.hpp`.

**Comment by abisner** [comment]
Same with `Cajita::P2G::is_scatter_view<T>`.

**Comment by streeve** [comment]
@junghans @dalg24, added the option to disable warnings in the last commit here, let me know if you have any other suggestions. @lebuller @kwitaechong @abisner can one of you also review?

**Review by junghans — APPROVED** [review]

File: cmake/CabanaConfig.cmakein [hunk 2]
```diff
@@ -21,10 +21,11 @@ set(Cabana_ENABLE_ARBORX @Cabana_ENABLE_ARBORX@)
...
   find_dependency(ArborX REQUIRED)
 endif()
-set(Cabana_ENABLE_CAJITA @Cabana_ENABLE_CAJITA@)
+set(Cabana_ENABLE_GRID @Cabana_ENABLE_GRID@)
+set(Cabana_ENABLE_CAJITA @Cabana_ENABLE_CAJITA@) # FIXME: deprecated
 include("${CMAKE_CURRENT_LIST_DIR}/Cabana_Targets.cmake")
-if(Cabana_ENABLE_CAJITA)
-  include("${CMAKE_CURRENT_LIST_DIR}/CajitaTargets.cmake")
+if(Cabana_ENABLE_GRID)
+  include("${CMAKE_CURRENT_LIST_DIR}/CabanaGrid_Targets.cmake")
```
[the remark this answers is not in the document]
Comment (streeve) [reply cmake/CabanaConfig.cmakein:29]: I do like the idea of using APPEND here - I'll give it a try

[the remark this answers is not in the document]
Comment (streeve) [reply cmake/CabanaConfig.cmakein]: I was looking in the wrong place, but the last commit uses the main Targets file for both

**Comment by kwitaechong** [comment]
> @junghans @dalg24, added the option to disable warnings in the last commit here, let me know if you have any other suggestions. @lebuller @kwitaechong @abisner can one of you also review?

Updated Picasso with new Cabana::Grid passes all tests. Looks good!

**Review by kwitaechong — APPROVED** [review]

**Comment by streeve** [comment]
Ignoring HIP timeout

**Pull request merged**